fastapi/dependencies/utils.py

```diff
@@ -412,7 +413,11 @@ def analyze_param(
             type_annotation
         ) or is_uploadfile_sequence_annotation(type_annotation):
             field_info = params.File(annotation=use_annotation, default=default_value)
-        elif not field_annotation_is_scalar(annotation=type_annotation):
+        elif (
+            not field_annotation_is_scalar(type_annotation)
+            # Root models by default regarded as bodies
```

Comment:
Backward compatibility. Root models have been regarded as body params by default eventhough they were scalar.

Reply:
Another option would be to explicitly require annotating by `Body`/`Query`,etc